### PR TITLE
CP-22609: Create Beta Chart Github Action 

### PR DESCRIPTION
**Pull request by bdrennz**
### Description

This PR adds a new Github Action which allows users to create a Beta tag, similar to how `latest` release tags are generated.

### Testing

TBD

### Checklist

- [x] I have added documentation for new/changed functionality in this PR
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: .github/workflows/build-and-publish-beta-chart.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        continue-on-error: true
+
+      # Step 8: Create GitHub Release
+      - name: Create Release
+        uses: softprops/action-gh-release@v2
+        with:
+          name: ${{ env.NEW_VERSION }}
+          tag_name: ${{ env.NEW_VERSION }}
+          files: cloudzero-agent-${{ env.NEW_VERSION }}.tgz
+          make_latest: ${{ github.event.inputs.release_type != 'beta' }}
```
Comment (bdrennz) [inline .github/workflows/build-and-publish-beta-chart.yml:148]: For anyone confused by this line:

If github.event.inputs.release_type is 'beta', the condition github.event.inputs.release_type != 'beta' evaluates to false, so make_latest is set to false. We don't want beta tags set to latest.

File: .github/workflows/build-and-publish-chart.yml [hunk 2]
```diff
@@ -50,7 +54,6 @@ jobs:
       - name: Build Dependencies
```
Comment (josephbarnett) [inline .github/workflows/build-and-publish-chart.yml:54]: Line 40-49 above assume we are merging from develop to main... which means any beta functionality is merged into main. Do we really want that? Vs trying to make this work from a feature-XXXX branch?

**Review by josephbarnett — COMMENTED** [review]
Looking closer at that workflow we might want to make a few changes:
1. Do we really want to merge to main?
2. How do we prevent from publishing the gh-pages... with the version info for betas?

File: .github/workflows/build-and-publish-chart.yml (hunk 2 above)
Comment (conradcz) [reply .github/workflows/build-and-publish-chart.yml:54]: I'll second this. Any subsequent merges to `main` that yield a stable version will include the "beta" changes from a prior pre-release version, which isn't ideal.

Comment (bdrennz) [reply .github/workflows/build-and-publish-chart.yml:59]: Added a condition that only merges when the user specifies to do so: https://github.com/Cloudzero/cloudzero-charts/pull/84/commits/c85a8c373295c39654544bac2c59234b69561dd8

**Comment by bdrennz** [comment]
@josephbarnett 

> 1. Do we really want to merge to main?

No we don't. The latest commit conditionally merges to main depending on the users input.


> How do we prevent from publishing the gh-pages... with the version info for betas?

Is there a reason we don't want this? Personally I like the idea of beta changes being published, users who find the beta chart beneficial can take advantage of the changes with the understanding that not all the functionality may work. A great example would be the deprecation of Node Exporter with the work we're currently doing. I'm definitely not partial to this, but seems like a nice way to get beta testers out of the box.

File: .github/workflows/build-and-publish-chart.yml (hunk 2 above)
Comment (josephbarnett) [reply .github/workflows/build-and-publish-chart.yml:59]: @bdrennz looking over this file - I think there are other locations which are impacted (ie. gh-pages). Would we want to maybe make a `clone` of this workflow focused only on feature branches?

Comment (bdrennz) [reply .github/workflows/build-and-publish-chart.yml:59]: > Would we want to maybe make a clone of this workflow focused only on feature branches?

Ya I think that would be the best option, we can always consolidate later.


> looking over this file - I think there are other locations which are impacted (ie. gh-pages).

I think it's worth considering some of the benefits I mention in my [comment](https://github.com/Cloudzero/cloudzero-charts/pull/84#issuecomment-2417854590). But if you have strong opinions against, I don't mind removing that step.

**Comment by josephbarnett** [comment]
> @josephbarnett
> 
> > 1. Do we really want to merge to main?
> 
> No we don't. The latest [commit](https://github.com/Cloudzero/cloudzero-charts/pull/84/commits/c85a8c373295c39654544bac2c59234b69561dd8) conditionally merges to main depending on the users input.
> 
> > How do we prevent from publishing the gh-pages... with the version info for betas?
> 
> Is there a reason we don't want this? Personally I like the idea of beta changes being published, users who find the beta chart beneficial can take advantage of the changes with the understanding that not all the functionality may work. A great example would be the deprecation of Node Exporter with the work we're currently doing. I'm definitely not partial to this, but seems like a nice way to get beta testers out of the box.

I think the challenge is that `beta` changes prevent ensuring `main` is only accepted/proven quality. I don't think we want them published in main. It makes getting them back out harder and defeats the purpose of a feature branch. 

That said, looking at this workflow - I see a few challenges with that suggestion. Specifically around ensuring the feature branch is accessible via the index file in the gh-pages page... kind of a harder thing to solve for....

**Pull request closed without merging**